### PR TITLE
LPS-117503 Fixes invalid date error

### PR DESCRIPTION
**Pull request by matuzalemsteles**
Just a small fix that was breaking the Forms DatePicker with "Invalid Date" still uses `moment` in some parts maybe it is an improvement to change it later, I didn’t change it because I may lose some case that date-fns doesn’t cover or the behavior is different compared to the moment in the date checks and does not make this ticket too big.

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
ci:test:sf

**Comment by matuzalemsteles** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ const DatePicker = ({
 					}
 
 					if (moment(value).isValid()) {
-						onChange(moment(value).format(dateMask));
+						onChange(moment(value).format(dateMask.toUpperCase()));
```
Comment (bryceosterhaus) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:193]: Good idea with this.

I think ideally we might swap out moment for date-fns here in the future

Comment (matuzalemsteles) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:193]: Yeah, I'm seeing how we can try to track this, we may have to create an epic to remove the use of moment from DXP.

**Comment by john-co** [comment]
This could be an opportunity to include a unit test to catch this kind of error in DatePicker so it doesn't trip us up on the short term? Maybe to check the date is stored with correct format "day/yyyy".

**Comment by matuzalemsteles** [comment]
> This could be an opportunity to include a unit test to catch this kind of error in DatePicker so it doesn't trip us up on the short term? Maybe to check the date is stored with correct format "day/yyyy".

Yeah, I imagine that unit tests already existed for this but this behavior is only when this component is used in the Forms context, I think I can verify that the component's `onChange` is emitting correctly, it should be enough.

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by wincent** [comment]
Screenshot 2020-07-22 -135144-HdVCD9dO@2x

(Base-switching trick to get rid of display of extraneous commits.)

**Comment by matuzalemsteles** [comment]
Hey guys, I will send forward this. I've just been looking at the tests that were breaking before the fix to see if everything is fine.

- LocalFile.Forms#SubmitFormWithAllFieldTypesAndThreePages - OK
- LocalFile.Forms#SubmitFormAfterEditingFields - OK
- FormRules#ConfigureRequireRule - OK

**Comment by matuzalemsteles** [comment]
ci:forward

**Pull request closed without merging**